### PR TITLE
Fixes for new exploit involving " in info strings.

### PR DESCRIPTION
**Pull request by In-line**
Related to #595 .

### CODE REVIEW

**Comment by In-line** [comment]
It was really pain to simplify Info_IsValid (removing redundant copies, copy-paste and so on). Behavior shouldn't be changed, so I didn't put refactoring under `REHLDS_FIXES`

**Comment by WPMGPRoSToTeMa** [comment]
What about reverting #595?

File: rehlds/engine/sv_main.cpp [hunk 1]
```diff
@@ -1979,6 +1979,7 @@ int SV_CheckForDuplicateNames(char *userinfo, qboolean bIsReconnecting, int nExc
 {
 	int dupc = 0;
 	char rawname[MAX_NAME];
+	char newname[MAX_NAME];
```
Comment (In-line) [inline rehlds/engine/sv_main.cpp]: OMG. REALLY? BECAUSE OF THIS BUILD FAILED?

**Comment by In-line** [comment]
@WPMGPRoSToTeMa Any suggestions?

**Pull request merged**